include/h2o/configurator.h

```diff
@@ -40,7 +40,11 @@ enum {
 
 #define H2O_CONFIGURATOR_NUM_LEVELS 4
 
-typedef struct st_h2o_configurator_context_t {
+typedef struct st_h2o_configurator_context_t h2o_configurator_context_t;
+
+typedef int(*h2o_configurator_filter_cb)(h2o_configurator_context_t *ctx, yoml_t **node);
```

Comment:
I doubt if passing in a pointer to a pointer to `yoml_t` is a good idea. I believe that the filter callback should update the contents of `yoml_t` instead of altering the address. The reason is as follows.

In current model, configurators do not _own_ the nodes. In other words, configurators do not call `yoml_free` upon exit.

But if you are going to allow an update of pointer to `yoml_t` (by passing `yoml_t **`), then every caller of the filter function would need to do some extra work so that it would call `yoml_free` taking the updated pointer as an argument.

My understanding is that this is not done at the moment, and also that we would not want to such thing.